common/src/account.rs

```diff
@@ -11,6 +11,12 @@ pub struct Account {
     pub userdata: Vec<u8>,
     /// contract id this contract belongs to
     pub program_id: Pubkey,
+
+    /// this account contains a program (and is strictly read-only)
+    pub executable: bool,
```

Comment:
@aeyakovenko had wanted to use an enum over `userdata` rather than adding a separate `executable` flag, which does seem much nicer

Reply:
Enum instead of Boolean, I'd guess

Reply:
Yeah, on the list